sdk/src/main/scala/com/horizen/AbstractWallet.scala

```diff
@@ -102,6 +105,22 @@ abstract class AbstractWallet[
 
   def applyConsensusEpochInfo(epochInfo: ConsensusEpochInfo): W
 
+  def generateNextSecret[T <: Secret](secretCreator: SecretCreator[T]): Try[W] = Try {
```

Comment:
This method should return a Tryof pair of (W, T), the caller need the generated secret

Reply:
Done